Task/133 implement control panel for authorized costcentres

p-kyllonen opened the pull request:
#133 Control panel for authorized costcentres

- List of all costcentres
- Search and sort costcentres
- Add new costcentres
- Modify costcentres
- Changelogs for costcentres

p-kyllonen committed:
(#133) Add costcentre  control panel

p-kyllonen committed:
(#133) Add costcentre  control panel

p-kyllonen committed:
(#133) Table data with sorting and searching

p-kyllonen committed:
(#133) Table data with sorting and searching

p-kyllonen committed:
(#133) Valid costcentre search and date fix

p-kyllonen committed:
(#133) Valid costcentre search and date fix

p-kyllonen committed:
(#133) Costcentre creation

p-kyllonen committed:
(#133) Costcentre creation

p-kyllonen committed:
(#133) Basic details of costcentres

p-kyllonen committed:
(#133) Basic details of costcentres

p-kyllonen committed:
(#133) Minor table fixes

p-kyllonen committed:
(#133) Minor table fixes

p-kyllonen committed:
(#133) Costcentre editing and date fixes

p-kyllonen committed:
(#133) Costcentre editing and date fixes

p-kyllonen committed:
(#133) Subject and changelog api change

p-kyllonen committed:
(#133) Subject and changelog api change

p-kyllonen committed:
(#133) Costcentre changelogs and some UI fixes

p-kyllonen committed:
(#133) Costcentre changelogs and some UI fixes

p-kyllonen committed:
Merge branch 'task/133-implement-control-panel-for-authorized-costcentres' of https://github.com/virkarekisteri/virkarekisteri into task/133-implement-control-panel-for-authorized-costcentres

p-kyllonen committed:
(#133) Unifying costcentre and subject tabs

p-kyllonen committed:
(#133) Added some comments

p-kyllonen committed:
(#133) Small changes to subject control panel

p-kyllonen committed:
(#133) Added some more comments

github-actions[bot] committed:
Fix code style issues with ESLint

github-actions[bot] committed:
Fix code style issues with Prettier

p-kyllonen committed:
(#133) ESLint error fix

p-kyllonen committed:
Merge branch 'task/133-implement-control-panel-for-authorized-costcentres' of https://github.com/virkarekisteri/virkarekisteri into task/133-implement-control-panel-for-authorized-costcentres

p-kyllonen committed:
(#133) ESLint error fix

p-kyllonen committed:
(#133) ESLint prop validation fix

Peskarhu commented:
(1) Voimassa alkaen ei saisi olla myöhemmin kuin voimassa päättyen päivämäärä
(2) Vain yksi numero, pitäisi neljä numeroa

![kuva](https://github.com/user-attachments/assets/7c17e9be-ed10-4ea1-93f7-c5875b159524)

p-kyllonen committed:
(#133) Costcentre number validation fix, date validation and missing sv translations

github-actions[bot] committed:
Fix code style issues with Prettier

p-kyllonen committed:
(#133) Date validation for modifying costcentre

p-kyllonen committed:
Merge branch 'task/133-implement-control-panel-for-authorized-costcentres' of https://github.com/virkarekisteri/virkarekisteri into task/133-implement-control-panel-for-authorized-costcentres

github-actions[bot] committed:
Fix code style issues with Prettier

Commit:
Small changes

* In backend, allow null values to update date. Fixes issue with
  updating or modifying a date with null value which was possible when
  creating in UI but not possible in modify.
* Dont allow until date to be the same as from date. Might make no
  difference, but also might make future date handling easier if not two
  same dates.
* Fix date formatting crash issue in editing history
  - Shows hourly time as 0.00.00 which might be good to fix in the
    future

Peskarhu reviewed (approved):
Vaikutti hyvältä ja toimivalta. UI näytti myös hienolta! Hyvää duunia
